### PR TITLE
few changes and Update baycMarketPlace.sol

### PR DESCRIPTION
**Pull request by dattaraj24**

### CODE REVIEW

**Comment by AngSin** [comment]
Thank you for the changes, ser. Much welcome 🙏
Hoping to stay in touch if you like, here: https://twitter.com/ultrasupahotfir

File: baycMarketPlace.sol [hunk 1]
```diff
@@ -163,12 +165,13 @@ contract baycMarketPlace {
             if (bid.bidder == _address) {
                 require(isApecoinSpendAllowed(_address, bid.bidInApecoin), "The bidder has not approved enough ApeCoin for this bid!");
                 require(IERC20(apecoinContract).balanceOf(_address) > bid.bidInApecoin, "The bidder does not have enough ApeCoin for this bid!");
+                
 
                 // accept bid
                 uint256 apecoinLeft = burnAndStoreApecoin(_address, bid.bidInApecoin);
+                delete tokenIdToBids[_tokenId][i];
```
Comment (AngSin) [inline baycMarketPlace.sol:172]: Before I merge, a question - why did you move this delete up? I assume the IERC20 and IERC721 token transfers can fail so that's why I had the delete afterwards, but maybe that is wrong practice. Or it probably does not matter since if they fail the whole transaction is reverted right?

**Comment by dattaraj24** [comment]
This is to avoid retrancy attack always make whatever state changes you
want to make before a transfer of the value otherwise other contracts can
manipulate the even before the value is set through sending multiple
transcations through another contract
